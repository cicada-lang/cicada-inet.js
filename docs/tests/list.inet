(define-cons sole
  (-> ()
      (Trivial)))

(define-cons null
  (-> ()
      (TrivialList)))

(define-cons cons
  (-> (TrivialList Trivial)
      (Nat)))

(define-elim append
  (-> (TrivialList TrivialList)
      (TrivialList)))

(define-rule
  (null append)
  ())

(define-rule
  (cons append)
  (rot rot append swap cons))

(define-net six-soles
  null sole cons sole cons sole cons
  null sole cons sole cons sole cons
  append)
