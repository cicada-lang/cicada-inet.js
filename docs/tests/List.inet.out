(cons₁₃)-value!target-(append₁₄)
(sole₁)-value head-(cons₂)
(null₀)-value tail-(cons₂)
(sole₃)-value head-(cons₄)
(cons₂)-value tail-(cons₄)
(sole₅)-value head-(cons₆)
(cons₄)-value tail-(cons₆)
(sole₈)-value head-(cons₉)
(null₇)-value tail-(cons₉)
(sole₁₀)-value head-(cons₁₁)
(cons₉)-value tail-(cons₁₁)
(sole₁₂)-value head-(cons₁₃)
(cons₁₁)-value tail-(cons₁₃)
(cons₆)-value rest-(append₁₄)
(append₁₄)-return

(sole₂₇)-value head-(cons₃₂)
(sole₂₀)-value head-(cons₃₄)
(cons₃₄)-value tail-(cons₃₂)
(sole₂₅)-value head-(cons₃₆)
(cons₃₆)-value tail-(cons₃₄)
(sole₁₈)-value head-(cons₃₈)
(cons₃₈)-value tail-(cons₃₆)
(sole₂₃)-value head-(cons₄₀)
(cons₄₀)-value tail-(cons₃₈)
(sole₁₆)-value head-(cons₄₂)
(cons₄₂)-value tail-(cons₄₀)
(null₁₅)-value tail-(cons₄₂)
(cons₃₂)-value

